### DISCUSSION TITLE
Welcome to OpenVehicleDiag Discussions!

### DISCUSSION THREAD

**Discussion by rnd-ash** [rnd-ash/OpenVehicleDiag#6] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zulu6** [comment]
Hi Ash,

Really like what you're up to!

I also have an older Mercedes (2006) that needs K-line in order for me to work on it.
I have Vediamo & Xentry but the J2534 devices I've ordered through Alibaba have failed repeatedly; hence I'm really interested in the Macchina.
I have a backup set of computers to test on (SAM, ECU, and Ebox).
Any update on how things are coming along with the k-line part of the J2534 drivers for Win?

All dressed up with nowhere to go (yet),


E

**Reply by rnd-ash** [reply]
hmmm, what 2006 Mercedes do you have?
If you have a C/E class, you should be able to use my M2 driver for ISO-TP with powertrain ECUs (TCM, ECU, ESP) (That is what I use on my C200CDI).

No update for K-Line yet, I have been swamped with University work lately so i might put that off until i can get my report out of the way.

**Comment by zulu6** [comment]
Great to hear from you, Ash.
I have a Mercedes-Smart Fortwo cdi from 2006; apparently the brainchild of a German-Swiss collaboration notoriously overloaded with safety features (I think of it as an airbag on wheels).
The forums seem to say you need to change comms in Vediamo to K-line to access the MEG and SAM units properly; I haven't made it that far due to hardware problems.

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by RussellA118** [comment]
Hi Ash,
I must say great work mate, I am very keen to see this all develop.
Just to add to your youtube video on opendiag your were talking about actual values and .S files
I know i have seen for example if you do an actuation or a test in DAS and the pre conditions for the test are say.. coolant temp must be above 80 degrees before the test can start. those values are stored in the .S files if i rememeber correctly i was able to edit the value.
Also Take a look into the .cnt files they seems to store all the routines for coding.
there is a function on das where you can start a K-Line debug and also a CSD debugger where you "step" through das.
If you like I can send you the .S and .CNT text files for your EGS52

**Reply by rnd-ash** [reply]
Yes i am aware of this. Check my new project [openstar](https://github.com/rnd-ash/openStar)...This is meant to be a special software suite compatible with DAS/Xentry diagnostic software files. So lots of reverse engineering atm with CAL/CSD.dll!

**Reply by RussellA118** [reply]
Oh yes! I am following your openstar, I shall be testing it in the workshop.
Any chance of a youtube video or live stream while you reverse engineer?

**Reply by rnd-ash** [reply]
Yes that is the plan for next week:
https://youtu.be/7hxznZGTj48

**Comment by KilveEng** [comment]
Hi Ash, been having a catchup on your opendiag software and interested on getting it to work on OSX, (may need some help) but mainly where in the uk can we buy the Macchina A0? or would I be better getting a M2? I need CAN access
Shane

**Comment by gobftald** [comment]
May be I'm a bit late, but I'm not able to find a Maacchina device to buy. All of those devices show that they are out of stock. No any news about the company in the last 2 years. So, without an open and programable HW device your OVD initiative a bit hard to follow/implement. Do you have any idea or access to any other device which can be used a similar way. Even with lego-ing from several cheap components. If you have some vision about the future of your project please share. I can even help with programming new devices and device drivers.
